internal/services/servicenetworking/application_load_balancer_association_resource.go

```diff
@@ -0,0 +1,230 @@
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+
+		"application_load_balancer_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: associationsinterface.ValidateTrafficControllerID,
+		},
+
+		"subnet_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: commonids.ValidateSubnetID,
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (t AssociationResource) Attributes() map[string]*schema.Schema {
+	return map[string]*schema.Schema{}
+}
+
+func (t AssociationResource) ModelObject() interface{} {
+	return &AssociationModel{}
+}
+
+func (t AssociationResource) ResourceType() string {
+	return "azurerm_application_load_balancer_association"
+}
+
+func (t AssociationResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return associationsinterface.ValidateAssociationID
+}
+func (t AssociationResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			trafficControllerClient := metadata.Client.ServiceNetworking.TrafficControllerInterface
+			client := metadata.Client.ServiceNetworking.AssociationsInterface
+
+			var config AssociationModel
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding %v", err)
+			}
+
+			parsedTrafficControllerId, err := associationsinterface.ParseTrafficControllerID(config.ApplicationLoadBalancerId)
+			if err != nil {
+				return err
+			}
+
+			controllerId := trafficcontrollerinterface.NewTrafficControllerID(parsedTrafficControllerId.SubscriptionId, parsedTrafficControllerId.ResourceGroupName, parsedTrafficControllerId.TrafficControllerName)
+			controller, err := trafficControllerClient.Get(ctx, controllerId)
+			if err != nil {
+				return fmt.Errorf("retrieving %s: %+v", controllerId, err)
+			}
+
+			if controller.Model == nil {
+				return fmt.Errorf("retrieving %s: Model was nil", controllerId)
+			}
+
+			loc := controller.Model.Location
+
+			id := associationsinterface.NewAssociationID(parsedTrafficControllerId.SubscriptionId, parsedTrafficControllerId.ResourceGroupName, parsedTrafficControllerId.TrafficControllerName, config.Name)
+			existing, err := client.Get(ctx, id)
+			if err != nil {
+				if !response.WasNotFound(existing.HttpResponse) {
+					return fmt.Errorf("checking for presence of exisiting %s: %+v", id, err)
+				}
+			}
+
+			if !response.WasNotFound(existing.HttpResponse) {
+				return tf.ImportAsExistsError(t.ResourceType(), id.ID())
+			}
+
+			association := associationsinterface.Association{
+				Location: location.Normalize(loc),
+				Properties: &associationsinterface.AssociationProperties{
+					Subnet: &associationsinterface.AssociationSubnet{
+						Id: config.SubnetId,
+					},
+					AssociationType: associationsinterface.AssociationTypeSubnets,
```

Comment:
Is there more association types that subnet? If that is the case, we probably want to rename the resource to something like `azurerm_application_load_balancer_association_subnet`?

Reply:
Till now, it's the only type

Reply:
Can we further confirm with the service team to see whether they have plan to support more type?

Reply:
still waiting for confirm, but updated other codes